src/tests/simulation/tests/integration_tests/backfill_tests.rs

```diff
@@ -264,7 +264,7 @@ async fn test_arrangement_backfill_progress() -> Result<()> {
     let progress = progress.replace('%', "");
     let progress = progress.parse::<f64>().unwrap();
     assert!(
-        (1.0..2.0).contains(&progress),
```

Comment:
I got 9.9 in sim test. Not sure if I got something broken. cc @kwannoel

Reply:
Yeah, I think it's because of the following sequence of events:
1. On barrier we refresh the snapshot read stream.
2. We also reconstruct the rate limiter.
3. IF barrier gets emitted at every 250ms, this makes rate limit ineffective, since we reset the rate limit every 250ms.

The fix for this is simple. If rate limit doesn't change, don't recreate the rate limiter and just reuse the old one.

Reply:
- https://github.com/risingwavelabs/risingwave/pull/15948/commits/6ad717ef8e48bb174538793e8b240c8ff6a35d0a
- https://github.com/risingwavelabs/risingwave/pull/15948/commits/ea341bd48a81eb83338142621e5b0f2aa4bcae24

Fixed no shuffle and arrangement backfill.